File: spa/SmartReminder/Models/Moment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SmartReminder.Models
{
    public class Moment
    {   
        public int Id
        { 
            get; 
            set; 
        }

        public string Name
        {
            get;
            set;
        }

        public int EventId
        {
            get;
            set;
        }

        public Event Event
        {
            get
            {
                return (Event)this.EventId;
            }
            set
            {
                this.EventId = (int)value;
            }
        }

        public Wifi Wifi
        { 
            get; 
            set; 
        }

        public DateTime StartTime
        { 
            get; 
            set; 
        }

        public DateTime EndTime
        { 
            get; 
            set; 
        }

        public DayOfWeek[] Days
        {
            get;
            set;
        }
    }

    public enum Event
    {
        WifiConnected,
        WifiDisconnected,
        ChagerConnected,
        ChargerDisconnected
    }
}

File: SmartReminder/src/com/example/smartreminder/models/Moment.java
package com.example.smartreminder.models;

import java.util.Date;
import java.util.List;

import com.example.smartreminder.Action;

public class Moment
{
	public String name;
 
	public Action action;
	
	public String extra;
	
	public String startTime;
	
	public String endTime;
	
	public Date specificDate;
	
	public List<Integer> days;
}
